### DISCUSSION TITLE
SBTN in CI

### DISCUSSION THREAD

**Discussion by JR-Utily** [sbt/sbt#7308] [Q&A] [answered]
Hello !
We have automated some build in our jenkins server, and it can build our code from several branches at the same time.
Thus, we are not sure if it would be safe to use sbt --client option, we are afraid one build would invalidate the result of a concurrent one, or worst, that the target artefact would be inconsistent.
Any insight on this topic ?

**Comment by eed3si9n** [comment, marked as answer]
In general it's not safe to run two sessions of sbt concurrently, if they point to the same mutable directories `target/` etc on a disk. This does not change regardless of sbtn or not.

One situation that I have speculated but not confirmed is that on Jenkins agent, you could potentially end up with having sbt server up and running, _sequentially_ pointing to different branches. I am not sure if you meant that. That situation in fact can potentially cause issues.

In theory:

```scala
Global / onChangedBuildSource := ReloadOnSourceChanges
```

should be able to detect that `build.sbt` has changed so you can try that. Otherwise you might need to add

```bash
sbt --client shutdown
```

**Reply by JR-Utily** [reply]
Thank you !
Every build should be in a different workspace so it should be safe, even if the files are almost the same.
